shared/@hedvig-ui/Button/button.tsx

```diff
@@ -205,20 +207,27 @@ const ButtonIconWrapper = styled.div`
 `
 
 export const Button = styled(
-  ({
-    icon,
-    children,
-    ...props
-  }: ButtonProps & ButtonHTMLAttributes<HTMLButtonElement>) => {
-    return (
-      <button {...props}>
-        <ButtonIconWrapper>
-          {!!icon && <ButtonIcon size={props.size}>{icon}</ButtonIcon>}
-          <div>{children}</div>
-        </ButtonIconWrapper>
-      </button>
-    )
-  },
+  React.forwardRef(
+    (
+      { icon, children, focus, ...props }: ButtonProps,
+      ref: React.ForwardedRef<HTMLButtonElement>,
+    ) => {
+      const internalRef = useRef<HTMLButtonElement>(null)
+      useElementFocus(
+        (ref as React.RefObject<HTMLElement>) ?? internalRef,
+        focus || false,
+      )
+
+      return (
+        <button {...props} ref={ref ?? internalRef}>
```

Comment:
Do something like this instead

```suggestion
  React.forwardRef(
    (
      { icon, children, focus, ...props }: ButtonProps,
      forwardRef: React.ForwardedRef<HTMLButtonElement>,
    ) => {
      const internalRef = useRef<HTMLButtonElement>(null)
      const ref = forwardRef ?? internalRef
      
      useElementFocus(
        ref,
        focus || false,
      )

      return (
        <button {...props} ref={ref}>
```